src/main/java/com/synopsys/integration/alert/ApplicationConfiguration.java

```diff
@@ -136,4 +138,23 @@ public AuthorizationManager authorizationManager(RoleAccessor roleAccessor) {
         return new AuthorizationManager(roleAccessor);
     }
 
+    @Bean
+    public DefaultJmsListenerContainerFactory defaultJmsListenerContainerFactory(CachingConnectionFactory connectionFactory) {
+        DefaultJmsListenerContainerFactory factory = new DefaultJmsListenerContainerFactory();
+        factory.setConnectionFactory(connectionFactory);
+        return factory;
+    }
+
+    @Bean
+    public DefaultJmsListenerContainerFactory distributionChannelJmsListenerContainerFactory(CachingConnectionFactory connectionFactory) {
+        DefaultJmsListenerContainerFactory factory = new DefaultJmsListenerContainerFactory();
+        factory.setConnectionFactory(connectionFactory);
+        int minConcurrency = 1;
+        int maxThreadCount = Runtime.getRuntime().availableProcessors();
+        int maxConcurrency = maxThreadCount < 10 ? maxThreadCount : 10;
```

Comment:
Once these beans are in separate classes, can we create a constant for this? `MAX_ALLOWED_THREAD_COUNT`. Then we should also include a comment on why this number was chosen.